spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkInputPartition.java

```diff
@@ -46,12 +47,14 @@ class SparkInputPartition implements InputPartition, HasPartitionKey, Serializab
       Types.StructType groupingKeyType,
       ScanTaskGroup<?> taskGroup,
       Broadcast<Table> tableBroadcast,
+      String branch,
```

Comment:
The input partition could also take the table schema instead of branch, but it will cause additional overhead for serialization, so I kept this to still pass in branch.